partridge/gtfs.py

```diff
@@ -191,6 +173,32 @@ def get(self, filename):
 
         return df
 
+    @contextmanager
+    def _io_adapter(self, filename):
+        """
+        Yield an IO object for the given file
+        from a zip file or folder
+        """
+        if self.is_dir:
+            iowrapper = open(self.zmap[filename], 'rb')
+
+            def close():
+                iowrapper.close()
+        else:
+            zipreader = ZipFile(self.path)
+            zfile = zipreader.open(self.zmap[filename], 'r')
+            iowrapper = io.TextIOWrapper(zfile, encoding='utf-8-sig')
+
+            def close():
+                iowrapper.close()
+                zfile.close()
+                zipreader.close()
+
+        try:
+            yield iowrapper
+        finally:
+            close()
```

Comment:
why not use the natural ctx managers for `open` and `ZipFile`? seems like it might be less fragile to rely on them to manage their own close behavior.

Reply:
Hmmm does that work with Python 2 & 3?

Reply:
afaik yeah. everything in `io` has had a standard contextmanager interface forever and `ZipFile` is heavily inspired by it.